server/src/main/java/org/opensearch/common/xcontent/XContentHelper.java

```diff
@@ -135,9 +134,9 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
                         convertToMap(XContentFactory.xContent(contentType), raw, offset, length, ordered));
             } else {
                 input = bytes.streamInput();
-                contentType = xContentType != null ? xContentType : XContentFactory.xContentType(input);
             }
             try (InputStream stream = input) {
+                contentType = xContentType != null ? xContentType : XContentFactory.xContentType(stream);
```

Comment:
`contentType` from else-if block seems to be overwritten. Can you check if this is desired?

Reply:
Oh.. Good catch. I should not change the current logic in any case..

Reply:
However, I don't think I changed the program logic, and `contentType` from the else-if block will be returned in L133, without to be overwritten. Could you help me double check?